packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -498,8 +498,8 @@ class HotRunner extends ResidentRunner {
 
   void _resetDirtyAssets() {
     for (final FlutterDevice? device in flutterDevices) {
-      device!.devFS!.assetPathsToEvict.clear();
-      device.devFS!.shaderPathsToEvict.clear();
+      device!.devFS?.assetPathsToEvict.clear();
```

Comment:
Changing the `!` to a `?` here was the fix. I'm not sure why the devFS is null here.